parachain/Cargo.toml

```diff
@@ -18,6 +18,7 @@ sp-io = { git = "https://github.com/paritytech/substrate", branch = "cumulus-bra
 lazy_static = { version = "1.4.0", optional = true }
 parking_lot = { version = "0.10.0", optional = true }
 log = { version = "0.4.8", optional = true }
+polkadot-primitives = { path = "../primitives", default-features = false }
```

Comment:
I don't think we want to bring `polkadot-primitives` in as a dependency. It depends on substrate primitives crates that we don't want to force all parachains to link to.

Reply:
The motivation there was to enable the [`use polkadot_primitives::BlockNumber` import](https://github.com/paritytech/polkadot/pull/879/commits/b4fb14bc521d270096b28df0d752cc174e934cb0#diff-3502f76737488b8321d5eef08588afa3R59). If we replace the block number with the relay chain header, we still have to either `use polkadot_primitives::Header`, or make the [`ValidationParams` struct](https://github.com/paritytech/polkadot/pull/879/commits/b4fb14bc521d270096b28df0d752cc174e934cb0#diff-3502f76737488b8321d5eef08588afa3L63-R78) generic. 

I think you're right: if we have a generic `ValidationParams`, then we don't need most of the refactor in https://github.com/paritytech/polkadot/pull/879/commits/b4fb14bc521d270096b28df0d752cc174e934cb0, which reduces the changes necessary to make all this work.

Reply:
I don't really like the idea of `ValidationParams` being generic, since we know what the types are going to be and are just getting around some type-fu.

I'd suggest instead breaking down `polkadot-primitives` into multiple crates.